src/main/java/com/apaliakou/homework04/homeworkcard/card/impl/CreditCard.java

```diff
@@ -0,0 +1,23 @@
+package com.apaliakou.homework04.homeworkcard.card.impl;
+
+import com.apaliakou.homework04.homeworkcard.card.Card;
+import java.math.BigDecimal;
+
+public class CreditCard extends Card {
+
+    public CreditCard(String holderName, BigDecimal balance) {
+        super(holderName, balance);
+    }
+
+    public CreditCard(String holderName) {
+        super(holderName);
+    }
+
+    @Override
+    public Card withdrawBalance(BigDecimal withdrawSum) {
+        checkInputSumForNegativity(withdrawSum);
```

Comment:
this check is already implemented in the src/main/java/com/apaliakou/homework04/homeworkcard/card/Card.java